Lib/test/test_interpreters.py

```diff
@@ -0,0 +1,88 @@
+import interpreters
+import os
+from textwrap import dedent
+import unittest
+
+def _captured_script(script):
+    r, w = os.pipe()
+    indented = script.replace('\n', '\n                ')
+    wrapped = dedent(f"""
+        import contextlib
+        with open({w}, 'w') as spipe:
+            with contextlib.redirect_stdout(spipe):
+                {indented}
+        """)
+    return wrapped, open(r)
+
+def clean_up_interpreters():
+    for id in interpreters.list_all():
+        if id == 0:  # main
+            continue
+        try:
+            interpreters.destroy(id)
+        except RuntimeError:
+            pass  # already destroyed
+
+
+class TestBase(unittest.TestCase):
+
+    def tearDown(self):
+        clean_up_interpreters()
+
+
+class ListAllTests(TestBase):
+
+    def test_initial(self):
```

Comment:
This is a good functional test.  I recommend indicating specifically in the name that this is a functional test.  Also please add a unit test.  Here are examples using the testing double from above:

```python
    def test_typical(self):
        stub = LowLevelStub()
        stub.return_list_all = [[0, 1, 3, 7]]

        found = interpreters.list_all(_interpreters=stub)

        self.assertEqual(found, [
            interpreters.Interpreter(0),
            interpreters.Interpreter(1),
            interpreters.Interpreter(3),
            interpreters.Interpreter(7),
            ])
        stub.check_calls(self, [
            ('list_all', (), {}),
            ])

    def test_minimum(self):
        stub = LowLevelStub()
        stub.return_list_all = [[0]]
        found = interpreters.list_all(_interpreters=stub)

        self.assertEqual(found, [
            interpreters.Interpreter(0),
            ])
        stub.check_calls(self, [
            ('list_all', (), {}),
            ])
```